microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -100,6 +119,44 @@ def verify_vmsnapshot_extension(
                     pass
                 else:
                     raise e
-            time.sleep(1)
+            time.sleep(60)
             count = count + 1
         assert_that(count, "Restore point creation failed.").is_less_than(10)
+
+    def get_restore_point(
+        self,
+        log: Logger,
+        environment: Environment,
+        resource_group_name: str,
+        restore_point_collection: str,
+        restore_point: str,
+    ) -> int:
+        assert environment.platform
+        platform: AzurePlatform = environment.platform  # type: ignore
+        assert isinstance(platform, AzurePlatform)
+        client = get_compute_client(platform)
+        attempts = 0
+        max_attempts = 2
+        while attempts < max_attempts:
+            response = client.restore_points.get(
+                resource_group_name=resource_group_name,
+                restore_point_collection_name=restore_point_collection,
+                restore_point_name=restore_point,
+                expand=None,
+            )
+            if response.provisioning_state == "Succeeded":
+                log.info(f"restore point {restore_point} created")
+                consistency_mode = response.consistency_mode
+                log.info(f"consistency mode is {consistency_mode}")
+                if (
+                    "FileSystemConsistent" in consistency_mode
+                    or "ApplicationConsistent" in consistency_mode
+                ):
+                    return 0
+                else:
+                    return 1
+            else:
+                log.info(f"rp status is {response.provisioning_state}")
+                time.sleep(900)
```

Comment:
we can check status in a loop, exit the loop till meet some condition, instead of hard code wait time

Reply:
So in this case we wanted to check the status of restore point creation after 15 minutes. and still it did not succeed then we would be returning 2 to fail the test.

Reply:
we should check the status in a loop, set max wait time as 15 mins, when status is expected, then exit the loop.